alibi/prototypes/protoselect.py

```diff
@@ -0,0 +1,543 @@
+        X_ft.append(preprocess_fn(X[istart:iend]))
+
+    return np.concatenate(X_ft, axis=0)
+
+
+def _imscatterplot(x: np.ndarray,
+                   y: np.ndarray,
+                   images: np.ndarray,
+                   figsize: Tuple[int, int],
+                   image_size: Tuple[int, int] = (28, 28),
+                   zoom: Optional[np.ndarray] = None,
+                   zoom_lb: float = 1.0,
+                   zoom_ub=2.0,
+                   sort_by_zoom: bool = True) -> None:
+    """
+    2D image scatter plot.
+
+    Parameters
+    ----------
+    x
+        Images' x-coordinates.
+    y
+        Images' y-coordinates.
+    images
+        Array of images to be placed at coordinates `(x, y)`.
+    figsize
+        `Matplotlib` figure size.
+    image_size
+        Size of the generated output image as `(rows, cols)`.
+    zoom
+        Images' zoom to be used.
+    zoom_lb
+        Zoom lower bound. The zoom values will be scaled linearly between `[zoom_lb, zoom_up]`.
+    zoom_ub
+        Zoom upper bound. The zoom values will be scaled linearly between `[zoom_lb, zoom_up]`.
+    """
+    if zoom is None:
+        zoom = np.ones(len(images))
+
+    zoom_min, zoom_max = np.min(zoom), np.max(zoom)
+    zoom = (zoom - zoom_min) / (zoom_max - zoom_min) * (zoom_ub - zoom_lb) + zoom_lb
+
+    if sort_by_zoom:
+        idx = np.argsort(zoom)[::-1]  # type: ignore
+        zoom = zoom[idx]  # type: ignore
+        x, y, images = x[idx], y[idx], images[idx]
+
+    fig, ax = plt.subplots(figsize=figsize)
+    ax.set_xticks([])
+    ax.set_yticks([])
+
+    resized_imgs = [resize(images[i], image_size) for i in range(len(images))]
+    imgs = [OffsetImage(img, zoom=zoom[i], cmap='gray') for i, img in enumerate(resized_imgs)]  # type: ignore
+    artists = []
+
+    for i in range(len(imgs)):
+        x0, y0, im = x[i], y[i], imgs[i]
+        ab = AnnotationBbox(im, (x0, y0), xycoords='data', frameon=False)
+        artists.append(ax.add_artist(ab))
+
+    ax.update_datalim(np.column_stack([x, y]))
+    ax.autoscale()
+
+
+def visualize_prototypes(explanation: 'Explanation',
+                         refset: Tuple[np.ndarray, np.ndarray],
+                         reducer: Callable[[np.ndarray], np.ndarray],
+                         preprocess_fn: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+                         figsize: Tuple[int, int] = (10, 10),
+                         image_size: Tuple[int, int] = (28, 28),
+                         zoom_lb: float = 1.0,
+                         zoom_ub: float = 3.0):
+    """
+    Plot the images of the prototypes at the location given by the `reducer` representation.
+    The size of each prototype is proportional to the log of the number of correct-class training images covered
+    by that prototype (Bien and Tibshiran (2012): https://arxiv.org/abs/1202.5933).
+
+    Parameters
+    ----------
+    explanation
+        Explanation object.
```

Comment:
Not just any `Explanation` object, but one that is of type `ProtoSelect`. Also ties in with discussion above whether this method should be an explainer or not.

Reply:
Added clarification, but still need to discuss